src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -24,34 +30,87 @@
     private Encoder m_leftEncoder = new Encoder(Constants.LEFT_ENCODER_PORTS[0], Constants.LEFT_ENCODER_PORTS[1]);
     private Encoder m_rightEncoder = new Encoder(Constants.RIGHT_ENCODER_PORTS[0], Constants.RIGHT_ENCODER_PORTS[1]);
 
+    private DifferentialDriveOdometry odometry;
+
+    private AHRS navX;
+
     public DriveTrain() {
         
         m_rightMotors.setInverted(true);
         m_differentialDrive = new DifferentialDrive(m_leftMotors, m_rightMotors);
         m_differentialDrive.setSafetyEnabled(false);
+
+        m_leftEncoder.setDistancePerPulse(Constants.ENCODER_DISTANCE_PER_PULSE);
+        m_rightEncoder.setDistancePerPulse(Constants.ENCODER_DISTANCE_PER_PULSE);
+
+        odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(0));
+        navX = new AHRS(Port.kMXP, (byte) 200);
     }
 
-    public double getRightSpeed() {
+    //Get the speed of the motors
+    public double getRightSpeed() {  
         return -m_rightMotors.get();
```

Comment:
Is this really what you want? Check the API. What "speed" are you really looking for?